output/cloud/expv2/flush.go

```diff
@@ -37,9 +41,23 @@ func (f *metricsFlusher) Flush(ctx context.Context) error {
 	msb := newMetricSetBuilder(f.referenceID, f.aggregationPeriodInSeconds)
 	for i := 0; i < len(buckets); i++ {
 		msb.addTimeBucket(&buckets[i])
+		if len(msb.seriesIndex) < f.maxSeriesInSingleBatch {
+			continue
+		}
```

Comment:
I was bit weary of an if in this loop ... but to be honest `addTimeBucket` is so much more complicated that this is likely not an issue.

If it is we can always optimize it.